pydomcfg/domzgr/sco.py

```diff
@@ -0,0 +1,199 @@
+            md96: bottom control parameter (0<= pbott <=1)
+            sf12: scaling factor for computing thickness
+            of bottom level Zb
+        alpha: float, optional
+            sf12: stretching parameter
+        efold: float, optional
+            sf12: efold length scale for transition from sigma
+            to stretched coord
+        pbot2: float, optional
+            sf12 offset for calculating Zb = H*pbott + pbot2
+        ln_e3_dep: bool
+            Logical flag to comp. e3 as fin. diff. (True) or
+            analyt. (False) (default = True)
+
+        Returns
+        -------
+        Dataset
+           Describing the 3D geometry of the model
+        """
+
+        self._bot_min = bot_min
+        self._bot_max = bot_max
+        self._hc = hc
+        self.rmax = rmax
+        self._stretch = stretch
+        self._ln_e3_dep = ln_e3_dep
+
+        # set stretching parameters after checking their consistency
+        if self._stretch:
+            self._set_stretch_par(psurf, pbott, alpha, efold, pbot2)
+
+        ds = self._init_ds()
+
+        # compute envelope bathymetry
+        ds_env = self._compute_env(ds)
+
+        # compute sigma-coordinates for z3 computation
+        kindx = ds_env["z"]
+        sigma = (self._compute_sigma(kk) for kk in kindx)
+        self._sigT, self._sigW = sigma
+
+        # compute z3 depths of zco vertical levels
+        # dsz = self._sco_z3(ds_env)
+
+        # compute e3 scale factors
+        # dse = self._compute_e3(dsz) if self._ln_e3_dep else self._analyt_e3(dsz)
+
+        # return dse
+
+    # --------------------------------------------------------------------------
+    def _set_stretch_par(self, psurf, pbott, alpha, efold, pbot2):
+        """
+        Set stretching parameters after checking
+        consistency of input parameters
+        """
+        if not (psurf and pbott):
+            if self._stretch == "sh94":
+                srf = "rn_theta"
+                bot = "rn_bb"
+            elif self._stretch == "md96":
+                srf = "rn_theta"
+                bot = "rn_thetb"
+            elif self._stretch == "sf12":
+                srf = "rn_zs"
+                bot = "rn_zb_a"
+            msg = (
+                srf
+                + " and "
+                + bot
+                + "MUST be set when using "
+                + self._stretch
+                + " stretching."
+            )
+            raise ValueError(msg)
+
+        if self._stretch == "sf12":
+            if not (alpha and efold and pbot2):
+                msg = "rn_alpha, rn_efold and rn_zb_b MUST be set when \
+                       using sf12 stretching."
+                raise ValueError(msg)
+
+        # setting stretching parameters
+        self._psurf = psurf if psurf else 0.0
+        self._pbott = pbott if pbott else 0.0
+        self._alpha = alpha if alpha else 0.0
+        self._efold = efold if efold else 0.0
+        self._pbot2 = pbot2 if pbot2 else 0.0
```

Comment:
```suggestion
        self._psurf = psurf or 0
        self._pbott = pbott or 0
        self._alpha = alpha or 0
        self._efold = efold or 0
        self._pbot2 = pbot2 or 0
```
Also, I think you can do this directly in `__call__`